internal/healthchecks/api_check.go

```diff
@@ -66,19 +56,16 @@ func createMonitoringPingRequest(gceMetadata resourcedetector.GCEResource) *moni
 		},
 	}
 	req := &monitoringpb.CreateTimeSeriesRequest{
-		Name: "projects/" + gceMetadata.Project,
+		Name: "projects/" + resource.GetProject(),
 		TimeSeries: []*monitoringpb.TimeSeries{{
 			MetricKind: metricpb.MetricDescriptor_GAUGE,
 			ValueType:  metricpb.MetricDescriptor_INT64,
 			Metric: &metricpb.Metric{
 				Type: metricType,
 			},
 			Resource: &monitoredres.MonitoredResource{
```

Comment:
If you make the interface change I suggested, this can just become `Resource: resource.MonitoredResource()`